homeassistant/components/risco/binary_sensor.py

```diff
@@ -82,15 +88,71 @@ def device_class(self):
         """Return the class of this sensor, from BinarySensorDeviceClass."""
         return BinarySensorDeviceClass.MOTION
 
-    async def _bypass(self, bypass):
-        alarm = await self._risco.bypass_zone(self._zone_id, bypass)
-        self._zone = alarm.zones[self._zone_id]
-        self.async_write_ha_state()
-
     async def async_bypass_zone(self):
         """Bypass this zone."""
         await self._bypass(True)
 
     async def async_unbypass_zone(self):
         """Unbypass this zone."""
         await self._bypass(False)
+
+    async def _bypass(self, bypass):
+        raise NotImplementedError
+
+
+class RiscoCloudBinarySensor(RiscoEntity, RiscoBinarySensor):
+    """Representation of a Risco cloud zone as a binary sensor."""
+
+    def __init__(self, coordinator, zone_id, zone):
+        """Init the zone."""
+        RiscoEntity.__init__(self, coordinator)
+        RiscoBinarySensor.__init__(self, zone_id, zone)
+
+    def _get_data_from_coordinator(self):
+        self._zone = self.coordinator.data.zones[self._zone_id]
+
+    @property
+    def unique_id(self):
+        """Return a unique id for this zone."""
+        return binary_sensor_unique_id(self._risco, self._zone_id)
+
+    async def _bypass(self, bypass):
+        alarm = await self._risco.bypass_zone(self._zone_id, bypass)
+        self._zone = alarm.zones[self._zone_id]
+        self.async_write_ha_state()
+
+
+class RiscoLocalBinarySensor(RiscoBinarySensor):
+    """Representation of a Risco local zone as a binary sensor."""
+
+    def __init__(self, system_id, zone_id, zone, zone_updates):
+        """Init the zone."""
+        super().__init__(zone_id, zone)
+        self._system_id = system_id
+        self._zone_updates = zone_updates
+
+    async def async_added_to_hass(self):
+        """Subscribe to updates."""
+        self._zone_updates[self._zone_id] = self.async_write_ha_state
+
+    @property
+    def extra_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            **super().extra_state_attributes,
+            "alarmed": self._zone.alarmed,
+            "groups": self._zone.groups,
```

Comment:
What do these state attributes represent? We try to avoid adding extra state attributes as far as possible.

Reply:
`alarmed` is whether this zone is currently causing an alarm, `groups` are the arming groups the zone belongs to. The former can arguably be a separate binary sensor, the latter is fairly static (requires reprogramming the alarm, which typically only installers have access to).
We already have `bypassed`, which today would have been a switch, but requires some restructuring (and is a breaking change), so I suggest I split off `bypassed` and `alarmed` to separate entities in a future PR.

Reply:
How will the users use the `groups` info?

We shouldn't add `alarmed` now if we plan to make it a binary sensor. Either make it a binary sensor now or later.

Reply:
* I imagine `groups` will most likely be consumed in the UI as a guide when building automations, e.g. if I arm group C, will I need to bypass this zone first?
BTW the fact that we're exposing groups is really cool - the Risco app doesn't show them anywhere, which means people can't even understand how their home is configured.
* I'll probably drop `alarmed` for now and add it in a future PR where I also break `bypassed` out to a switch. Need to first figure out the best way to have multiple entities subscribe to the same zone update.

Reply:
If the user wants to bypass a zone in the automation based on which group is armed, shouldn't the info relationship be the other way around? Ie we expose the zone entity ids that belong to an alarm group on the alarm entity that controls the alarm group. Then it's easy to check the state attributes of the alarm that is targeted in the automation and it should work to bypass all those zone entities programmatically in the automation.

Reply:
There are no entities for groups. Groups are just sets of zones, but partitions are the units that are represented by individual alarm control panels (Risco terminology is not ideal).
To add to the complexity, groups span across partitions so are not part of a hierarchy (zones can also belong to multiple partitions).

Reply:
Ok.